airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -184,11 +186,14 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
         if not self.cursor_field:
             yield from records
         else:
-            cursor_date = (stream_state or {}).get(self.cursor_field)
+            cursor_date = (stream_state or {}).get(self.cursor_field) or self._start_date
             for record in records:
                 updated = record[self.cursor_field]
                 if not cursor_date or updated > cursor_date:
+                    self._finished = False
                     yield record
+                else:
+                    self._finished = True
```

Comment:
@marcosmarxm Yes, it's handled by `check_start_time_param` method to prevent making API calls with `> now()`. 

Please give more context around `Any chance to a page with more frequent records be read before a old page?`.

Reply:
Is it possible to the connector request the `page=5` data before the `page=2`?

Reply:
Yes, this is possible, since the connector uses future_requests lib for some streams, that doesn't guarantee the order of operations.